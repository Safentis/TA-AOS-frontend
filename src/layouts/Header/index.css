.header {
  background-color: var(--main-dk-grey-color);
  border-bottom: 0.5rem solid var(--main-white-color);
}
