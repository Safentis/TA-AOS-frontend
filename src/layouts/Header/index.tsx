import React, { FC } from 'react';

import './index.css';
import { Props } from './index.interface';

const Header: FC<Props> = ({
  className = '',
}): React.ReactElement => {
  return (
    <header className={`header ${className}`}>
      <div className="header__inner">
        <h2 className="header__title">Rick and Morty</h2>
      </div>
    </header>
  );
};

export default Header;
