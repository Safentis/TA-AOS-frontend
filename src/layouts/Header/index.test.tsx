import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Header from './index';

import { Props } from './index.interface';

const setUp = (props: Props) => {
  return shallow(<Header {...props} />);
};

describe('<Header />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has root heade element', () => {
      expect(component.find('header')).toBeTruthy();
    });

    it('has root heade element', () => {
      expect(component.find('h2')).toBeTruthy();
      expect(
        component.find('h2').text() === 'Rick and Morty',
      ).toBeTruthy();
    });
  });
});
