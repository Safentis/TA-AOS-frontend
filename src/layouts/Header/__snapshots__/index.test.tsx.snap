// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component does match to the snapshot 1`] = `
<header
  className="header custom-style"
>
  <div
    className="header__inner"
  >
    <h2
      className="header__title"
    >
      Rick and Morty
    </h2>
  </div>
</header>
`;
